lib/ecto/query/builder.ex

```diff
@@ -215,6 +215,13 @@ defmodule Ecto.Query.Builder do
     {expr, params}
   end
 
+  def escape({op, _, _}, _type, _params, _vars, _env) when op in ~w(|| && !)a do
+    error! """
+          Short-form operators are not supported: `#{op}`
+          Instead use the full name as found in SQL: `and`, `or`, and `not`.
+          """
```

Comment:
I do agree that the message should be improved – my thoughts behind mentioning SQL was to explain the fact that it's treated as an error, as opposed to simply supporting them as aliases.